[importccl] improve import re-order unit test

C0rWin opened the pull request:
This commit refactors target-cols-reordered unit tests to include more
sophisticated example of import column reordering.

Signed-off-by: Artem Barger <bartem@il.ibm.com>

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45734)
<!-- Reviewable:end -->

C0rWin commented:
Addressing @dt review comment from PR #45713 regarding unit test.

C0rWin committed:
[importccl] improve import re-order unit test

This commit refactors target-cols-reordered unit tests to include more
sophisticated example of import column reordering.

Signed-off-by: Artem Barger <bartem@il.ibm.com>

Release note: none

C0rWin commented:
Closing this, since #45713 wasn't merged yet, so once re-opened would be possible to update UT directly there.
